sway-lsp/src/capabilities/runnable.rs

```diff
@@ -1,25 +1,62 @@
+use serde_json::Value;
 use sway_core::language::parsed::TreeType;
-use tower_lsp::lsp_types::Range;
+use tower_lsp::lsp_types::{Command, Range};
 
-#[derive(Debug, Eq, PartialEq, Hash)]
-pub enum RunnableType {
+#[derive(Debug, Eq, PartialEq, Hash, Clone, Copy)]
+pub enum RunnableKind {
     /// This is the main_fn entry point for the predicate or script.
     MainFn,
     /// Place holder for when we have in language testing supported.
     /// The field holds the index of the test to run.
-    _TestFn(u8),
+    TestFn(u8),
 }
 
-#[derive(Debug, Eq, PartialEq)]
+#[derive(Debug, Eq, PartialEq, Clone)]
 pub struct Runnable {
+    /// The kind of runnable
+    pub kind: RunnableKind,
     /// The location in the file where the runnable button should be displayed
     pub range: Range,
     /// The program kind of the current file
     pub tree_type: TreeType,
+    /// Additional arguments to use with the runnable command
+    pub arguments: Option<Vec<Value>>,
```

Comment:
Is it possible to document the specific values that are expected for arguments? Since Value can represent 6 unique types it's a bit unclear what is valid or not to send in here.

Reply:
Done! Now, there is a Runnable trait and 2 structs with implementations for Runnable. Each struct is responsible for implementing it's own `arguments`, making it really clear which arguments are needed.